alf/utils/plot_tb_curves.py

```diff
@@ -0,0 +1,240 @@
+            from scipy.signal import savgol_filter
+            curve = savgol_filter(curve, smoothing, polyorder=1)
+
+        return curve
+
+
+class EnvironmentStepsReturnReader(MeanCurveReader):
+    """Create a mean curve reader that reads AverageReturn values."""
+
+    def _get_metric_name(self):
+        return "Metrics_vs_EnvironmentSteps/AverageReturn"
+
+
+class EnvironmentStepsSuccessReader(MeanCurveReader):
+    """Create a mean curve reader that reads Success rates."""
+
+    def _get_metric_name(self):
+        return "Metrics_vs_EnvironmentSteps/success"
+
+
+class CurvesPlotter(object):
+    """Plot several MeanCurves in a figure. The curve colors will form
+    a cycle over 10 default colors.
+    """
+    _COLORS = ['C%d' % i for i in range(10)]
+    _LINE_WIDTH = 2
+
+    def __init__(self,
+                 mean_curves,
+                 x_range=None,
+                 x_label=None,
+                 y_label=None,
+                 title=None):
+        """
+        Args:
+            mean_curves (MeanCurve|list[MeanCurve]):
+            x_range (tuple[float]): a tuple of (min_x, max_x) for showing on
+                the figure. If None, then (0, 1) will be used.
+            x_label (str): shown besides x-axis
+            y_label (str): shown besides y-axis
+            title (str): title of the figure
+        """
+        self._fig, ax = plt.subplots(1)
+
+        if not isinstance(mean_curves, list):
+            mean_curves = [mean_curves]
+
+        if x_range is None:
+            x_range = (0., 1.)
+
+        n_points = len(mean_curves[0].y)
+        delta_x = (x_range[-1] - x_range[0]) / (n_points - 1)
+        x = np.arange(n_points) * delta_x + x_range[0]
+
+        for i, c in enumerate(mean_curves):
+            color = self._COLORS[i % len(self._COLORS)]
+            ax.plot(x, c.y, color=color, lw=self._LINE_WIDTH, label=c.name)
+            ax.fill_between(x, c.max_y, c.min_y, facecolor=color, alpha=0.3)
+
+        ax.legend(loc='upper left')
+        ax.grid(linestyle='--')
+
+        if x_label:
+            ax.set_xlabel(x_label)
+        if y_label:
+            ax.set_ylabel(y_label)
+        if title:
+            ax.set_title(title)
+
+    def plot(self, output_path, dpi=200, transparent=False):
+        """Plot curves and save the figure to disk.
+
+        Args:
+            output_path (str): the output file path
+            dpi (int): dpi for the figure. A higher value results in higher
+                resolution.
+            transparent (bool): If True, then the figure has a transparent
+                background.
+        """
+        self._fig.savefig(
+            output_path, dpi=dpi, transparent=transparent, bbox_inches='tight')
+        plt.close(self._fig)
+
+
+def _get_curve_path(dir):
+    return os.path.join(HOME, "tensorboard_curves", dir)
+
+
+if __name__ == "__main__":
+    """A plotting example."""
+    event_files = glob.glob(_get_curve_path("sac_kickball_gs/*/eval"))
```

Comment:
comment that the event_files are usually multiple runs of the same configuration with different random seeds.